src/web/app/components/teammates-common/submission-status-name.pipe.spec.ts

```diff
@@ -1,8 +1,53 @@
+import { FeedbackSessionSubmissionStatus } from '../../../types/api-output';
 import { SubmissionStatusNamePipe } from './submission-status-name.pipe';
 
 describe('SubmissionStatusNamePipe', () => {
   it('create an instance', () => {
     const pipe: SubmissionStatusNamePipe = new SubmissionStatusNamePipe();
     expect(pipe).toBeTruthy();
   });
+
+  it('transform with no deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasNoDeadlines = {
+      studentDeadlines: {},
+      instructorDeadlines: {},
+    };
+    const hasNoOngoingDeadlines = {
+      studentDeadlines: { nonOngoingExtension: new Date('2019-01-01').valueOf() },
+      instructorDeadlines: {},
+    };
+    const pipe: SubmissionStatusNamePipe = new SubmissionStatusNamePipe();
```

Comment:
Perhaps initialise this in a `beforeEach`?

Probably the same for `SubmissionStatusTooltipPipe`.